src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala

```diff
@@ -1,9 +1,15 @@
 package tv.codely.scala_intro_examples.lesson_08_collections
 
 final class WordAggregator {
-  // @ToDo: Here you have your exercise!
-  // @see tv.codely.scala_intro_examples.lesson_08_collections.WordAggregatorSpec
   def aggregateWords(text: String): Map[String, Int] = {
-    Map.empty
+    if (text.isEmpty)
+      return Map.empty
+    var occurrences = Map[String, Int]()
```

Comment:
Try to think about an alternative implementation without requiring mutability (`var`) 🙂

Hint: Instead of iterating over the `words: Array[String]` with a `foreach`, there're other methods which allow to accumulate a result between each array item iteration. Take a look at [the `scala.Array` documentation 😬](https://www.scala-lang.org/api/2.12.4/scala/Array.html)